### PR TITLE
Translation for Chapter 4

### PR DESCRIPTION
**Pull request by WUTONK**
Initial translation into Chinese, code comments to be processed

### CODE REVIEW

File: chapter04/chapter4.md [hunk 1]
```diff
@@ -340,10 +340,10 @@ glVertexAttribPointer(0, 3, GL_FLOAT, false, 0, 0);
 完成VBO后，我们可以解除它和VAO的绑定\(将它们绑定到0\)
 
 ```java
-// Unbind the VBO
+// Unbind the VBO 解除VBO绑定
```
Comment (Qubik65536) [inline chapter04/chapter4.md:343]: 注释要不只写中文要不只写英语

**Review by Qubik65536 — CHANGES_REQUESTED** [review]
完没完成？没有至少写Convert to draft

**Comment by Qubik65536** [comment]
@WUTONK 已转换成Draft，下次再有正在进行的工作自己去转

**Review by PencilCore — APPROVED** [review]
Hope you can decrease the blank lines between paragraphs, or not the book seems too overstaffed.

File: chapter04/chapter4.md [hunk 2]
```diff
@@ -1,30 +1,48 @@
...
 
-The sequence of steps that ends up drawing a 3D representation into your 2D screen is called the graphics pipeline. First versions of OpenGL employed a model which was called fixed-function pipeline. This model employed a set of steps in the rendering process which defined a fixed set of operations. The programmer was constrained to the set of functions available for each step. Thus, the effects and operations that could be applied were limited by the API itself \(for instance, “set fog” or “add light”, but the implementation of those functions were fixed and could not be changed\).
+在本章中，我们将学习使用OpenGL渲染场景时发生的过程和三维图形基本渲染过程。如果你习惯于旧版本的OpenGL，也就是固定的函数管道，您可能会在本章结束时想知道为什么它需要如此复杂。你可能会认为，只是在屏幕上画一个简单的形状不需要那么多的概念和代码。
+让我给你们这些有这种想法的人一个建议。它实际上更简单、更灵活。你只需要给它一个机：现代的OpenGL让你一次思考一个问题，它让你以一种更有逻辑的方式组织你的代码和进程。
```
Comment (PencilCore) [inline chapter04/chapter4.md]: > 你只需要给它一个机：

应是: `你只需要给它一个机会：`

**Comment by Qubik65536** [comment]
提交内包含不属于第四章的内容，还有，注释**只写中文**，**不要英语**

**Comment by Qubik65536** [comment]
不接受该Pull request

**Pull request closed without merging**